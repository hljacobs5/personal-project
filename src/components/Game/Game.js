import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import TriviaCard from '../TriviaCard/TriviaCard';
import './Game.css';

class Game extends Component {
	displayQuestions = () => {
		// if (this.props.pathname === '/politics') {
		return this.props.questions.results.map(result => {
			console.log(result)
			return <TriviaCard result={result}/>
		})
	  // } 
	}

	// handleSubmit() {

	// }

	render() {
		if (this.props.questions.results) {
		return (
			<div>
				{this.displayQuestions()}
				<button onClick={this.handleSubmit}>Submit</button>
			</div>
		)
	} else {
		return (
			<div>
				<h3>Not loaded</h3>
			</div>
		)
	}
  } 
}

export const mapStateToProps = (state) => ({
	questions: state.questions,
	answers: state.answers
})

export default connect(mapStateToProps)(Game);