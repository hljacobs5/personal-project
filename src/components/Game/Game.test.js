import React from 'react';
import { shallow, mount } from 'enzyme';
import { Game, mapStateToProps, mapDispatchToProps } from './Game';

describe('Game', () => {
	let wrapper;

	beforeEach(() => {
		const mockQuestions = { results: ['true', 'false']}
		wrapper = shallow(<Game questions={mockQuestions}
								answers={['true', 'true', 'false']}
								addScore={jest.fn()}
								history={[]}
						  />)
	})

	it('should exist', () => {
		expect(wrapper).toBeDefined();
	})

	it('should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})


	describe('handleSubmit', () => {
    	const mockEvent = { preventDefault: jest.fn() };

		it('should call handleSubmit on button click', () => {
			let mockHandleSubmit = jest.fn()
			wrapper.instance().handleSubmit = mockHandleSubmit
			wrapper.instance().forceUpdate()
			wrapper.find('button').simulate('click')
			expect(mockHandleSubmit).toHaveBeenCalled()
		})

		it('should call addScore under the right conditions', () => {
		
			wrapper.find('button').simulate('click')
			wrapper.instance().handleSubmit(mockEvent)

			expect(wrapper.instance().props.addScore).toHaveBeenCalled()
		})
	})
})