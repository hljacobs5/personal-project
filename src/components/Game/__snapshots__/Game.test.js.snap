// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Game
    addScore={[MockFunction]}
    answers={
      Array [
        "true",
        "true",
        "false",
      ]
    }
    history={Array []}
    questions={
      Object {
        "results": Array [
          "true",
          "false",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Connect(TriviaCard)
            result="true"
          />,
          <Connect(TriviaCard)
            result="false"
          />,
        ],
        <button
          onClick={[Function]}
        >
          Submit
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "result": "true",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "result": "false",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Connect(TriviaCard)
              result="true"
            />,
            <Connect(TriviaCard)
              result="false"
            />,
          ],
          <button
            onClick={[Function]}
          >
            Submit
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "result": "true",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "result": "false",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
